common/Geometry2d/Rect.hpp

```diff
@@ -63,6 +63,7 @@ class Rect : public Shape {
 
     Point center() const { return (pt[0] + pt[1]) / 2; }
 
+    /* Assumes that pt[0] <= pt[1] for both x and y */
```

Comment:
I'm not sure that we actually do make this assumption since this function is only used in one place where pt[0]==pt[1] at the start. I thought about that but wasn't sure what we wanted since our pts are mutable anyways so constructors can be bypassed. (I also realized this about the non=existent line asserts) idk if its worth since its such a specific case. (really I would prefer that this not be a function of point since its such an oddly specific use case where its used in the Polygon class).